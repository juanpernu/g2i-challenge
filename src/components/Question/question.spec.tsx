import { create } from 'react-test-renderer';
import { Question } from './index';

describe('<Question />', () => {
  const Mock = {
    question: {
      category: 'Test question',
      question: 'Is this a test question?',
      correct_answer: 'True',
      difficulty: 'hard',
      incorrect_answers: ['False'],
      type: 'boolean'
    },
    onClick: jest.fn()
  };

  it('should match snapshot', () => {
    const component = create(
      <Question question={Mock.question} onClick={Mock.onClick} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
