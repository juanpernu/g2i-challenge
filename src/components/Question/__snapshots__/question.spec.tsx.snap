// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> should match snapshot 1`] = `
<div
  className="p-10 bg-neutral-100 rounded-md shadow"
>
  <h2
    className="text-2xl font-semibold text-slate-800"
  >
    Test question
  </h2>
  <p
    className="text-slate-600 text-base my-2"
  >
    Is this a test question?
  </p>
  <div
    className="grid grid-cols-2 gap-4 mt-8"
  >
    <button
      aria-expanded="false"
      className="w-full bg-blue-600 rounded-md p-2 inline-flex items-center justify-center text-white hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-indigo-500"
      onClick={[Function]}
      type="button"
    >
      True
    </button>
    <button
      aria-expanded="false"
      className="w-full bg-blue-600 rounded-md p-2 inline-flex items-center justify-center text-white hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-indigo-500"
      onClick={[Function]}
      type="button"
    >
      False
    </button>
  </div>
</div>
`;
